modules/xfem/src/actions/XFEMAction.C

```diff
@@ -62,7 +62,8 @@ XFEMAction::act()
   MooseSharedPointer<XFEMInterface> xfem_interface = _problem->getXFEM();
   if (xfem_interface == NULL)
   {
-    MooseSharedPointer<XFEM> new_xfem (new XFEM(_problem->getMooseApp(), _problem));
+    _pars.set<FEProblem *>("_fe_problem") = &*_problem;
```

Comment:
@permcody what do you think of this? I'm getting a raw pointer to FEProblem from the reference to the shared pointer that we have in Action. I know that this defeats the smartness of the smart pointer, but there are raw pointers to FEProblem all over the place in MOOSE, and there's no ownership of that pointer in XFEM. Based on the recent observation by @jiangwen84, switching this over to a raw pointer probably fixes all of those valgrind errors. I'm building this on Linux right now to run with valgrind to see if that is the case.